packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart

```diff
@@ -669,16 +749,60 @@ class HeapTreeViewState extends State<HeapTree>
           ),
         ),
         const Spacer(),
-        Text(
-          controller.monitorTimestamp == null
-              ? 'No allocations tracked'
-              : 'Allocations Tracked at ${MemoryController.formattedTimestamp(controller.monitorTimestamp)}',
-          style: themeData.colorScheme.italicTextStyle,
+        updateCircle,
+      ],
+    );
+  }
+
+  Widget textWidgetWithUpdateCircle(
+    String text, {
+    TextStyle style,
+    double size,
+  }) {
+    final textWidth = textWidgetWidth(text, style: style);
+
+    return Stack(
+      children: [
+        Positioned(
+          child: Container(
+            child: Text(text, style: style),
+            width: textWidth + 10,
+          ),
+        ),
+        Positioned(
+          right: 0,
+          child: Container(
+            alignment: Alignment.topRight,
+            width: size,
+            height: size,
+            child: const Icon(Icons.circle, size: 0),
+            decoration: BoxDecoration(
+              shape: BoxShape.circle,
+              color: Colors.blue[400],
+            ),
+          ),
         ),
       ],
     );
   }
 
+  double textWidgetWidth(String message, {TextStyle style}) {
+    // self-defined constraint
+    const constraints = BoxConstraints(
+      maxWidth: 800.0, // maxwidth calculated
+    );
+    final richTextWidget = Text.rich(TextSpan(text: message), style: style)
+        .build(context) as RichText;
+    final renderObject = richTextWidget.createRenderObject(context);
```

Comment:
@grouma do you have some pointers to a better way to perform text measurement. Running the widget build pipeline is clever but a bit heavy weight. 
This is fine to leave for now but add a TODO.  This might cause some subtle bugs. For example, makes me nervous that we are passing our context to the createRenderObject call.

Reply:
Done. Let me know Gary if there is a lightweight computation?